posts/2025-12-23-investigating-spirv-for-the-shader-system-part-3.md

```diff
@@ -0,0 +1,343 @@
+---
+title: "Investigating SPIR-V for the shader system - Part 3"
+author: youness
+popular: false
+image: /images/spir.png
+tags: ['.NET', 'Shaders']
+---
+In this third part we're going to design a parser for the SDSL language!
```

Comment:
Can we make this intro a bit richer? This particular part is important for SEO, see your previous intros..

**Part 1:** In this first part of a new series of blog posts, we will learn more about Stride's shader system, its limitations and how to make it better thanks to a very useful shader language called SPIR-V. This will be the first step in implementing a new and better shader system.

**Part 2:**  In this second part we're going to dive deeper in how the current SDSL compiler works and how we are improving on it for the SPIR-V compiler. This will be a sort of personal log book of my research on the subject.